simulations/NavierStokes_2D/scripts/train_posteriori.jl

```diff
@@ -45,33 +47,75 @@ function loss_posteriori(model, p, st, data)
              sum(abs2, y))
 end
 
+NEPOCHS = 500
+LR = 1e-2
+NUSE_VAL = 2000
+
 # * train a-posteriori: old way single data point
 train_data_posteriori = dataloader_posteriori()
 optf = Optimization.OptimizationFunction(
-    (x, _) -> loss_posteriori(closure, x, st, train_data_posteriori), # x here is the optimization variable (θ params of NN)
+    (x, _) -> loss_posteriori(closure, x, st, dataloader_posteriori), # x here is the optimization variable (θ params of NN)
     Optimization.AutoZygote()
 )
 optprob = Optimization.OptimizationProblem(optf, θ)
-result_posteriori = Optimization.solve(
+optim_result, optim_t, optim_mem, _ = @timed Optimization.solve(
     optprob,
-    OptimizationOptimisers.Adam(0.1);
+    OptimizationOptimisers.Adam(LR);
     callback = callback,
-    maxiters = 100,
+    maxiters = NEPOCHS,
     progress = true
 )
-θ_posteriori = result_posteriori.u
+θ_posteriori_optim = optim_result.u
 
 # * package loss
 loss_posteriori_lux = create_loss_post_lux(dudt_nn2; sciml_solver = Tsit5())
 loss_posteriori_lux(closure, θ, st, train_data_posteriori)
 
 # * training via Lux
-loss, tstate = train(closure, θ, st, dataloader_posteriori, loss_posteriori_lux;
-    nepochs = 100, ad_type = Optimization.AutoZygote(),
-    alg = OptimizationOptimisers.Adam(0.1), cpu = true, callback = callback)
+lux_result, lux_t, lux_mem, _ = @timed train(
+    closure, θ, st, dataloader_posteriori, loss_posteriori_lux;
+    nepochs = NEPOCHS, ad_type = Optimization.AutoZygote(),
+    alg = OptimizationOptimisers.Adam(LR), cpu = true, callback = callback)
 
+loss, tstate = lux_result
 # the trained params are:
-θ_posteriori = tstate.parameters
+θ_posteriori_lux = tstate.parameters
+
+# Function to test the results
+function validate_results(p, dataloader, nuse = 100)
```

Comment:
I am wondering if this validation an plotting should be in this file or in `evaluate_posteriori.jl`, have you thought about that?

Reply:
Would be nice, however for the comparison we need to access the full training routine, while the idea of `evaluate_posteriori` is to just load a trained model. 

I think that this comparison should go to the [explorations](https://github.com/DEEPDIP-project/explorations) repo, but we include both loss functions to `src` and we show only the Lux one in `train_posteriori`. What do you think?

Reply:
Yes, I like the idea:
1. copy the script as it is to explorations (still with the loss defined in it for clarity). 
2. adding the code to `src/loss/loss_posteriori.jl`.